src/components/vmd-commune-or-departement-selector.component.ts

```diff
@@ -275,22 +243,25 @@ export class VmdCommuneOrDepartmentSelectorComponent extends VmdCommuneSelectorC
         const fullTextSearchableQuery = Strings.toFullTextSearchableString(this.filter)
 
         this.departementsAffiches = this.departementsCherchables.filter(dpt => {
-            return dpt.fullTextSearchableCodeDepartement.indexOf(fullTextSearchableQuery) === 0
-                || dpt.fullTextSearchableNom.indexOf(fullTextSearchableQuery) !== -1;
+            return dpt.fullTextSearchableCodeDepartement.startsWith(fullTextSearchableQuery)
```

Comment:
Je suis partagé sur `startsWith` et `includes` : ce sont des methodes d'Array qui peuvent rendre le code incompatible avec une très petite part de la population (je pense notamment [aux vieux Android 4.x](https://caniuse.com/mdn-javascript_builtins_string_startswith))

C'est pour ça que j'évitais de les utiliser au profit d' `indexOf()` (ça nous coûte pas grand chose d'utiliser `indexOf()`, juste un peu de lisibilité)

L'autre option serait d'utiliser des polyfills (basé sur https://github.com/es-shims/es-shim-api ?)

Reply:
ça coûte de la lisibilité à chaque fois que quelqu'un essaye de comprendre le code :)
https://caniuse.com/?search=startsWith dit que ça représente ~5% des navigateurs courants (y compris les chinois qu'on a pas chez nous) et qui sont déjà out pour plein d'autres raisons.
Après on peut utiliser un polyfill, parce que c'est cool comme fonctions =)

Reply:
je check pour un polyfill

Reply:
Je pense qu'utiliser des polyfill est la meilleure solution. Avec [polyfill.io](https://polyfill.io), il y a même moyen de ne les charger que si le navigateur est ancien. Cela marche via la détection du __user-agent__. Sinon il y a le plugin [legacy](https://github.com/vitejs/vite/tree/main/packages/plugin-legacy) de vite.

Reply:
got for vite legacy, c'est exactement ça qu'il faut :) ça sait lire une target list